src/ray/core_worker/context.h

```diff
@@ -129,15 +139,15 @@ class WorkerContext {
   // The runtime env info.
   std::shared_ptr<rpc::RuntimeEnvInfo> runtime_env_info_ GUARDED_BY(mutex_);
   /// The id of the (main) thread that constructed this worker context.
-  const boost::thread::id main_thread_id_;
+  const std::thread::id main_thread_id_ GUARDED_BY(mutex_);
```

Comment:
`boost:thread:id` is not hashable, and to be used as key for absl::flat_hash_map somehow....